### PR TITLE
Simplify ERC721Wrapper.depositFor to save gas

### PR DESCRIPTION
**Pull request by Amxx**
This PR reduces the gas cost of depositing tokens into a ERC721Wrapper (using the `depositFor` function) by about 3.2k gas per token. (see test report)

#### PR Checklist

- [x] Tests

### CODE REVIEW

**Comment by frangio** [comment]
Another simplification we have to make is remove `ERC721Holder` and inherit `IERC721Receiver` directly, since we're fully overriding it.

File: contracts/token/ERC721/extensions/ERC721Wrapper.sol [hunk 1]
```diff
@@ -63,17 +67,8 @@ abstract contract ERC721Wrapper is ERC721, ERC721Holder {
      * WARNING: Doesn't work with unsafe transfers (eg. {IERC721-transferFrom}). Use {ERC721Wrapper-_recover}
      * for recovering in that scenario.
      */
-    function onERC721Received(
-        address operator,
-        address from,
-        uint256 tokenId,
-        bytes memory data
-    ) public override returns (bytes4) {
+    function onERC721Received(address, address from, uint256 tokenId, bytes memory) public override returns (bytes4) {
```
Comment (Amxx) [inline contracts/token/ERC721/extensions/ERC721Wrapper.sol:70]:
```suggestion
    function onERC721Received(address, address from, uint256 tokenId, bytes memory) public virtual override returns (bytes4) {
```

**Review by frangio — APPROVED** [review]
LGTM

**Review by ernestognw — APPROVED** [review]
Thanks!

**Pull request merged**